onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -0,0 +1,571 @@
+
+          if (input_dims_size != input_arrays[i].size())
+            return false;
+        }
+      }
+
+      //Disable MNIST for MYRIAD and HDDL
+      if (dev_id == "MYRIAD" || dev_id == "HDDL")
+        return false;
+    }
+    //Dropout or Identity can't have graph inputs
+    if (node->OpType() == "Dropout" || node->OpType() == "Identity" || node->OpType() == "Concat") {
+      auto graph_inputs = graph_viewer.GetInputs();
+      for (auto input : node->InputDefs()) {
+        auto it = find(graph_inputs.begin(), graph_inputs.end(), input);
+        if (it != graph_inputs.end()) {
+          return false;
+        }
+      }
+    }
+
+    //Attribute auto pad for MaxPool and AveragePool must not be empty or SAME_LOWER
+    if (node->OpType() == "MaxPool" || node->OpType() == "AveragePool") {
+      auto attributes = node->GetAttributes();
+      auto auto_pad = attributes["auto_pad"].s();
+      if (auto_pad == "" || auto_pad == "SAME_LOWER"){
+        return false;
+      }
+
+      auto strides_ints = attributes["strides"].ints();
+      if(auto_pad == "SAME_UPPER" && strides_ints.size() == 0){
+          return false;
+      }
+
+      //Dilations have to be 1
+      auto dilations_ints = attributes["dilations"].ints();
+      if (dilations_ints.size() != 0) {
+        if (dilations_ints[0] > 1)
+          return false;
+      }
+
+      //Don't support ceil_mode = 1
+      auto ceil_mode = attributes["ceil_mode"].i();
+      if (ceil_mode != 0)
+        return false;
+
+      //Don't support multiple outputs for Pooling
+      if (node->OutputDefs().size() > 1) {
+        return false;
+      }
+    }
+
+    //Only support 4D and 5D blobs for CPU,GPU
+    //Only support 3D and 4D blobs for MYRIAD and HDDL
+    if (node->OpType() == "GlobalMaxPool" || node->OpType() == "MaxPool" || node->OpType() == "AveragePool" || node->OpType() == "GlobalAveragePool") {
+      auto graph_inputs = graph_viewer.GetInputs();
+      auto it = find(graph_inputs.begin(), graph_inputs.end(), node->InputDefs()[0]);
+      if (it != graph_inputs.end()) {
+        if (dev_id == "MYRIAD" || dev_id == "HDDL") {
+          if (input_dims != 3 || input_dims != 4)
+            return false;
+        } else if (input_dims < 4 || input_dims > 5) {
+          return false;
+        }
+      }
+    }
+
+    //Transpose with no attr is not supported
+    if (node->OpType() == "Transpose") {
+      auto attributes = node->GetAttributes();
+      auto perm = attributes["perm"].ints();
+      if (perm.size() == 0 || perm.size() > 5) {
+        return false;
+      }
+
+      //String data type is not supported
+      const auto* type_proto = node->InputDefs()[0]->TypeAsProto();
+      if (type_proto->tensor_type().elem_type() == ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_STRING) {
+        return false;
+      }
+
+      auto graph_inputs = graph_viewer.GetInputs();
+      auto it = find(graph_inputs.begin(), graph_inputs.end(), node->InputDefs()[0]);
```

Comment:
same comment as I had above. why does this logic only apply if node input is a graph input?

Reply:
Resolved